odoo/osv/expression.py

```diff
@@ -1429,107 +1481,114 @@ def __leaf_to_sql(self, leaf, model, alias):
         assert not isinstance(right, BaseModel), \
             "Invalid value %r in domain term %r" % (right, leaf)
 
-        table_alias = '"%s"' % alias
-
         if leaf == TRUE_LEAF:
-            query = 'TRUE'
-            params = []
+            return SQL("TRUE")
 
-        elif leaf == FALSE_LEAF:
-            query = 'FALSE'
-            params = []
+        if leaf == FALSE_LEAF:
+            return SQL("FALSE")
 
-        elif operator == 'inselect':
+        field = model._fields[left]
+        sql_field = model._field_to_sql(alias, left, self.query)
+
+        if operator == 'inselect':
             subquery, subparams = right
-            query = '(%s."%s" in (%s))' % (table_alias, left, subquery)
-            params = list(subparams)
+            return SQL("(%s IN (%s))", sql_field, SQL(subquery, *subparams))
 
-        elif operator == 'not inselect':
+        if operator == 'not inselect':
             subquery, subparams = right
-            query = '(%s."%s" not in (%s))' % (table_alias, left, subquery)
-            params = list(subparams)
+            return SQL("(%s NOT IN (%s))", sql_field, SQL(subquery, *subparams))
 
-        elif operator in ['in', 'not in']:
+        if operator in ('in', 'not in'):
             # Two cases: right is a boolean or a list. The boolean case is an
             # abuse and handled for backward compatibility.
             if isinstance(right, bool):
                 _logger.warning("The domain term '%s' should use the '=' or '!=' operator." % (leaf,))
                 if (operator == 'in' and right) or (operator == 'not in' and not right):
-                    query = '(%s."%s" IS NOT NULL)' % (table_alias, left)
+                    return SQL("(%s IS NOT NULL)", sql_field)
                 else:
-                    query = '(%s."%s" IS NULL)' % (table_alias, left)
-                params = []
+                    return SQL("(%s IS NULL)", sql_field)
+
+            elif isinstance(right, SQL):
+                sql_operator = SQL_OPERATORS[operator]
```

Comment:
Can you move this just before the `if operator in ('in', 'not in'):`, the same line is repeated 4 times (and the cost of the line is very low anyway).